lib/workers/repository/update/branch/reuse.ts

```diff
@@ -10,6 +10,26 @@ type ParentBranch = {
   isConflicted?: boolean;
 };
 
+async function shouldKeepUpdated(
+  config: BranchConfig,
+  baseBranch: string,
+  branchName: string,
+): Promise<boolean> {
+  const branchPr = await platform.getBranchPr(
+    config.branchName,
+    config.baseBranch,
+  );
+
+  if (branchPr) {
+    const keepUpdatedLabel = config.keepUpdatedLabel;
+    if (keepUpdatedLabel && !!branchPr.labels?.includes(keepUpdatedLabel)) {
+      return true;
+    }
+  }
```

Comment:
Refactor for performance so that the PR is only fetched if a `keepUpdatedLabel` is defined